### PR TITLE
Feature/database interfaces tests

### PR DESCRIPTION
**Pull request by Michalski-Artur**
- [x] Implement integration tests for DataAccess
- [x] Implement unit tests for OfferService 
- [ ] Change db provider so tests can be run on github

### CODE REVIEW

**Comment by Michalski-Artur** [comment]
Integration tests for DataAccess fail cause database for tests is ran on local sql server

**Comment by cyaneko** [comment]
I'll let this through for now, but this means we need either a way to deploy and test the database using GitHub Actions, or a remote test database (might happen with CD looming?)

**Review by cyaneko — APPROVED** [review]

**Comment by Michalski-Artur** [comment]
> I'll let this through for now, but this means we need either a way to deploy and test the database using GitHub Actions (perhaps using SQLite for tests only might be enough?) or a remote test database (might happen with CD looming?)

Honestly speaking I thought you will handle it on your own as you know best GitHub Actions @cyaneko

**Comment by cyaneko** [comment]
Sure, but we'll need to first know which solution we're going with
CD needs some templates which are not available atm
________________________________

**Review by mtabor99-pw — APPROVED** [review]

**Review by daveedoo — CHANGES_REQUESTED** [review]
Could you please elaborate what is your idea behind tests naming? If you had any special in mind.
I mean, I don't clearly understand why every test method starts with `Can_`.
I ask because it'd be good to have one common convention across the project.

File: Server-Module/Server.Tests/Services/OfferServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        public OfferServiceTest()
+        {
+            var config = new MapperConfiguration(opts =>
+            {
+                opts.AddProfile(new AutoMapperProfile());
+            });
+            _mapper = config.CreateMapper();
+            _dataAccessMock = new Mock<IDataAccess>();
+
+            _offerService = new OfferService(_dataAccessMock.Object, _mapper);
```
[the remark this answers is not in the document]
Comment (mtabor99-pw) [reply Server-Module/Server.Tests/Services/OfferServiceTest.cs:27]: Well, I for one think that naming private class members with an underscore at the beginning is helpful when it comes to differentiating between a local variable and a private class member - especially when method definition is long. Besides it helps when intellisense "filters" all private variables when you type an underscore.

**Comment by Michalski-Artur** [comment]
I altered some things due to your suggestions @daveedoo
- Change DataAccess to OfferDataAccess 
- Change tests name convention
- Add IDisposable
- Anonymous type creation is seperated into seperate lines
If you would be so kind to take a glance :)

**Comment by Michalski-Artur** [comment]
Tests naming convention:
- if any additional conditions are not present (ex. happy path): `[function_name]_[expected_behaviour]`
- if any additional conditions are present: `[function_name]_[additional_conditions]_[expected_behaviour]`

**Review by daveedoo — APPROVED** [review]
cool

**Pull request merged**